libs/amount/amount.cpp

```diff
@@ -64,7 +54,7 @@ namespace amount {
 
   Amount Amount::percentage(const Amount &am) const {
     uint256_t new_value = value_ * am.value_;
-    new_value /= uint256_t(std::pow(10, precision_ + am.precision_));
+    new_value /= ipow(10, am.precision_+2);
```

Comment:
magic number :( obvious from the fn name but still

Reply:
Added more comments, please check

Reply:
Thanks!